tests/shopify_python/test_google_styleguide.py

```diff
@@ -226,33 +226,76 @@ def test_use_cond_exprs(self):
         with self.assertNoMessages():
             self.walk(root_not_msg)
 
-    def test_unary_lambda_func(self):
+    def test_unary_lambda_func_with_complex_operand_allowed(self):
         unary_root = astroid.builder.parse("""
         def unaryfnc():
             unary_pass = map(lambda x: not (x+3), [1, 2, 3, 4])
-            unary_fail = map(lambda x: -x, [1, 2, 3, 4])
         """)
+        with self.assertNoMessages():
+            self.walk(unary_root)
+
+    @pytest.mark.parametrize('test_case', [
+        ('- x', 'neg'),
+        ('~ x', 'invert'),
+        ('not x', 'not_'),
+        ('+ x', 'pos')
+    ])
+    def test_unary_lambda_func(self, test_case):
+        (expression, op_name) = test_case
+        unary_root = astroid.builder.parse("""
+        def unaryfnc():
+            unary_fail = map(lambda x: {}, [1, 2, 3, 4])
+        """.format(expression))
```

Comment:
These changes appear to be built upon https://github.com/Shopify/shopify_python/pull/73; I assume that you'll rebase before merging.